project/shacl/modos_schema.shacl.ttl

```diff
@@ -126,291 +140,335 @@ modos:VariantSet a sh:NodeShape ;
-            sh:description "The URI from which a resource or dataset was obtained or derived." ;
-            sh:maxCount 1 ;
-            sh:order 1 ;
-            sh:path modos:source_uri ],
-        [ sh:datatype xsd:string ;
-            sh:description "A string specifying the release or version of a software or resource." ;
-            sh:maxCount 1 ;
-            sh:order 2 ;
-            sh:path modos:version ],
-        [ sh:datatype xsd:string ;
+    sh:property [ sh:datatype xsd:string ;
             sh:description "The pre-computed hash uniquely representing a biological sequence. Calculated as the MD5 of the upper-case sequence excluding all whitespace characters (this is equivalent to SQ:M5 in SAM)." ;
             sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 0 ;
             sh:path modos:sequence_md5 ;
             sh:pattern "^[a-f0-9]{32}$" ],
         [ sh:datatype xsd:string ;
             sh:description "A human-readable name for a thing" ;
             sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 4 ;
             sh:path schema1:name ],
+        [ sh:datatype xsd:string ;
+            sh:description "A string specifying the release or version of a software or resource." ;
+            sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
+            sh:order 2 ;
+            sh:path modos:version ],
+        [ sh:datatype xsd:anyURI ;
+            sh:description "A unique identifier for a thing" ;
+            sh:maxCount 1 ;
+            sh:minCount 1 ;
+            sh:nodeKind sh:Literal ;
+            sh:order 3 ;
+            sh:path schema1:identifier ],
         [ sh:datatype xsd:string ;
             sh:description "A human-readable description for a thing" ;
             sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 5 ;
-            sh:path schema1:description ] ;
+            sh:path schema1:description ],
+        [ sh:datatype xsd:anyURI ;
+            sh:description "The URI from which a resource or dataset was obtained or derived." ;
+            sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
+            sh:order 1 ;
+            sh:path modos:source_uri ] ;
     sh:targetClass modos:ReferenceSequence .
 
 modos:ReferenceGenome a sh:NodeShape ;
     sh:closed true ;
     sh:description "Reference assembly of a given genome, consisting of a collection of congiguous sequences (contigs)." ;
     sh:ignoredProperties ( rdf:type ) ;
-    sh:property [ sh:class modos:ReferenceSequence ;
-            sh:description "Denotes that a sequence belongs to a collection (e.g. a reference genome)." ;
-            sh:nodeKind sh:IRI ;
-            sh:order 1 ;
-            sh:path modos:has_sequence ],
+    sh:property [ sh:datatype xsd:string ;
+            sh:description "The path to access a resource, on a network or local filesystem. Can be a path relative to the root of the digital object, a URL, or an absolute path." ;
```

Comment:
you might be able to restrict this accurately

_thanks AI daddy_
`
        sh:or (
            # Case 1: Relative path
            [
                sh:pattern "^(?!/|https?://|file:///).+" ;
                sh:message "The value must be a relative path."
            ]
            # Case 2: URL (supports http, https, and file URLs)
            [
                sh:pattern "^(https?://|file:///).+" ;
                sh:message "The value must be a URL starting with http, https, or file."
            ]
            # Case 3: Absolute path (Unix-style absolute path)
            [
                sh:pattern "^/.*" ;
                sh:message "The value must be an absolute path starting with '/'."
            ]
        ) ;
    ] .
`